pkg/canary/deployment_controller.go

```diff
@@ -197,6 +197,19 @@ func (c *DeploymentController) ScaleFromZero(cd *flaggerv1.Canary) error {
 		if primary.Spec.Replicas != nil && *primary.Spec.Replicas > 0 {
 			replicas = primary.Spec.Replicas
 		}
+	} else if cd.Spec.AutoscalerRef != nil {
+		hpa, err := c.kubeClient.AutoscalingV2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
+		if err == nil {
+			if hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
+				replicas = hpa.Spec.MinReplicas
+			}
+		} else {
+			// fallback to v2beta2
+			hpa, err := c.kubeClient.AutoscalingV2beta2().HorizontalPodAutoscalers(cd.Namespace).Get(context.TODO(), cd.Spec.AutoscalerRef.Name, metav1.GetOptions{})
+			if err == nil && hpa.Spec.MinReplicas != nil && *hpa.Spec.MinReplicas > 1 {
```

Comment:
tiny nitpick: could you split this if condition to be similar to the one on L202 and L203?